app/controllers/api/v2/bookmarks_controller.rb

```diff
@@ -0,0 +1,222 @@
+        bookmark_url: found_result[:bookmark_url] || "",
+        bookmark_id: bookmark[:id],
+        original_url: bookmark[:url],
+        messages: found_result[:bookmark_messages] || ["No bookmark found for archivist '#{archivist.login}' and url '#{bookmark[:url]}'"]
+      )
+    end
+
+    messages = ["Successfully searched bookmarks for archivist '#{archivist.login}'"]
+    render_api_response(status, messages, bookmarks: results)
+  end
+
+  # POST - create a bookmark for this archivist
+  def create
+    archivist = User.find_by(login: params[:archivist])
+    bookmarks = params[:bookmarks]
+    bookmarks_responses = []
+    @bookmarks = []
+
+    # check for top-level errors (not an archivist, no bookmarks...)
+    status, messages = batch_errors(archivist, bookmarks)
+
+    if status == :ok
+      # Flag error and successes
+      @some_errors = @some_success = false
+
+      # Process the bookmarks
+      archivist_bookmarks = Bookmark.where(pseud_id: archivist.default_pseud.id)
+      bookmarks.each do |bookmark|
+        bookmarks_responses << create_bookmark(archivist, bookmark, archivist_bookmarks)
+      end
+
+      # set final response code and message depending on the flags
+      status = :bad_request if bookmarks_responses.any? { |r| [:ok, :created, :found].exclude?(r[:status]) }
+      messages = response_message(messages)
+    end
+
+    render_api_response(status, messages, bookmarks: bookmarks_responses)
+  end
+
+  private
+
+  # Find bookmarks for this archivist
+  def check_archivist_bookmark(archivist, original_url, archivist_bookmarks)
+    current_bookmark_url = original_url
+    archivist_bookmarks = archivist_bookmarks
+                            .select { |b| b.bookmarkable_type == "ExternalWork" && b.bookmarkable.url == current_bookmark_url }
+                            .map    { |b| [b, b.bookmarkable] }
+
+    if archivist_bookmarks.present?
+      archivist_bookmark, archivist_bookmarkable = archivist_bookmarks.first
+      find_bookmark_response(
+        bookmarkable: archivist_bookmarkable,
+        bookmark_status: :found,
+        bookmark_message: "There is already a bookmark for #{archivist.login} and the URL #{current_bookmark_url}",
+        bookmark_url: bookmark_url(archivist_bookmark)
+      )
+    else
+      find_bookmark_response(
+        bookmarkable: nil,
+        bookmark_status: :not_found,
+        bookmark_message: "There is no bookmark for #{archivist.login} and the URL #{current_bookmark_url}",
+        bookmark_url: ""
+      )
+    end
+  end
+
+  # Create a bookmark for this archivist using the Bookmark model
+  def create_bookmark(archivist, params, archivist_bookmarks)
+    found_result = {}
+    bookmark_request = bookmark_request(archivist, params)
+    bookmark_status, bookmark_messages = bookmark_errors(bookmark_request)
+    bookmark_url = nil
+    original_url = nil
+    bookmarkable = nil
+    @some_errors = true
```

Comment:
This is in the old code, so I assume it's fine, but could you explain why this is set to true?

Reply:
I suspect I did this because there's only one code path that indicates success (the one where `bookmark.save` succeeds), so it's simpler to only flip it in that one success case and leave it set to "yep, it didn't work" in all other cases.